Integration of Simtext toolset

dlal-group opened the pull request:
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

bgruening reviewed (commented):
Super cool! A few comments inline :)

* do you want to add a licence for your python scripts? You could do this by adding the license attribute to the tool-tag.
* the tool help could be improved a little bit we are using often ** what is does ** and similar markup, please have a look at other tools

bgruening commented on tools/simtext/pubmed_by_queries.xml:
```diff
@@ -0,0 +1,83 @@
+      ]]>
+    </command>
+
+    <environment_variables>
```
Saving this as an environment variable is dangerous as Galaxy is capturing those as part of the job output and exposes it, e.g. in shared histories etc ...

A better way is to save it to a file and pass that file over to your script. Please see here https://github.com/galaxyproject/tools-iuc/blob/master/tools/pyega3/pyega3.xml#L37 for an example.

bgruening reviewed (commented):
@dlal-group thanks for the update. I added one comment regarding the credentials.

Do you want to include a license for your R scripts? You could set a `license` https://docs.galaxyproject.org/en/latest/dev/schema.html#attributes

bgruening commented on tools/simtext/pubmed_by_queries.xml:
```diff
@@ -16,14 +16,16 @@
       $abstract
       #if $__user__.extra_preferences.get('ncbi_account|apikey', ""):
-        -k "\$NCBI_APIKEY"
+        -k $__user__.extra_preferences.get('ncbi_account|apikey', "")
```
```suggestion
        -k '$credentials'
```

bgruening commented on tools/simtext/pubmed_by_queries.xml:
```diff
@@ -16,14 +16,16 @@
-    </environment_variables>
+    <configfiles>
+        <configfile name="credentials"><![CDATA[
+            #set $ncbi_apikey = $__user__.extra_preferences.get('ncbi_account|apikey', "")
```
```suggstion
$__user__.extra_preferences.get('ncbi_account|apikey', "")
```

bgruening reviewed (commented):
I think you need something like that. Your R script needs to read from the credentials file. This way the credential is not exposed to the commandline.

dlal-group commented:
@bgruening thank you for the suggestion, we added a license to the tools. We also removed the environment variable. Using an API key in the preferences works for me, however I am not sure if the code is right like that? The interactive tool is added to the galaxy interactive tools in a pull request here: https://github.com/galaxyproject/galaxy/pull/11680
Thank you!

bgruening commented:
@dlal-group yeah, super cool please see my columns above.

dlal-group commented:
hi @bgruening I tried different ways/ syntax to get the API key from the credentials file but it does not work for me. it seems like the R script is running but no abstracts are retrieved. The same API works if it does not come from the intermediate file but directly as a variable. Can you help or do you know how to pass the variable from the credentials to the command? Thank you so much!

bgruening commented:
@dlal-group that looks correct to me, but you need to adjust your R script. You need to read the key from the file. The argument passed to `-k` is **not** a API key now, its a path to a file and from that file you need to read the API key.

bgruening commented on tools/simtext/pubmed_by_queries.R:
```diff
@@ -0,0 +1,257 @@
+# -o OUTPUT, --output OUTPUT  output file name. [default "pubmed_by_queries_output"]
+# -n NUMBER, --number NUMBER  number of PMIDs or abstracts to save per ID [default "5"]
+# -a, --abstract              if abstracts instead of PMIDs should be retrieved use --abstracts
+# -k KEY, --key KEY           if NCBI API key is available, add it to speed up the fetching of pubmed data
```
this help needs to be adjusted now

bgruening commented on tools/simtext/README.md:
```diff
@@ -0,0 +1,198 @@
+ -o OUTPUT, --output OUTPUT  output file name [default "pubmed_by_queries_output"]
+ -n NUMBER, --number NUMBER  number of PMIDs or abstracts to save per ID [default "5"]
+ -a, --abstract              if abstracts instead of PMIDs should be retrieved use --abstracts 
+ -k KEY, --key KEY           if NCBI API key is available, add it to speed up the download of PubMed data
```
this needs to be adjusted now

bgruening commented on tools/simtext/pubmed_by_queries.R:
```diff
@@ -0,0 +1,257 @@
+parser$add_argument("-a", "--abstract", action = "store_true", default = FALSE,
+                    help = "if abstracts instead of PMIDs should be retrieved use --abstracts ")
+parser$add_argument("-k", "--key", type = "character",
+                    help = "if ncbi API key is available, add it to speed up the download of pubmed data")
```
this is now a file

bgruening commented on tools/simtext/pubmed_by_queries.xml:
```diff
@@ -0,0 +1,85 @@
+
+**Info**
+
+Users can obtain an API key from the Settings page of their NCBI account (to create an account, visit http://www.ncbi.nlm.nih.gov/account/).
```
This API key needs to be added to the Galaxy user-preferences.

bgruening reviewed (approved):
This looks ready to me @dlal-group! Just a few small comments about help text and documentation.

bgruening commented:
Great! Danke Marie!
